src/erlcloud_config.erl

```diff
@@ -0,0 +1,370 @@
+-define(RESOURCE_TYPES, <<"ResourceTypes">>).
+-define(CONFIGURATION_RECORDER_NAMES, <<"ConfigurationRecorderNames">>).
+-define(DELIVERY_CHANNEL_NAMES, <<"DeliveryChannelNames">>).
+-define(CHRONOLOGICAL_ORDER, <<"chronologicalOrder">>).
+-define(EARLIER_TIME, <<"earlierTime">>).
+-define(LATER_TIME, <<"laterTime">>).
+-define(INCLUDE_DELETED_RESOURCES, <<"includeDeletedResources">>).
+
+
+%%------------------------------------------------------------------------------
+%% Library initialization.
+%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                config_host       = Host,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                config_host       = Host,
+                config_port       = Port,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer(), string()) ->
+    aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                config_host       = Host,
+                config_port       = Port,
+                config_scheme     = Scheme,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+%%------------------------------------------------------------------------------
+%% AWS Config API Functions
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% DescribeComplianceByConfigRule
+%%
+%% https://docs.aws.amazon.com/config/latest/APIReference/API_DescribeComplianceByConfigRule.html
+%%------------------------------------------------------------------------------
+-spec describe_compliance_by_config_rule(Opts :: map(),
+    Config :: aws_config()) ->
+    result_paged(map()).
+describe_compliance_by_config_rule(Opts, Config) when is_map(Opts) orelse Opts =:= #{} ->
+    Request = #{?COMPLIANCE_TYPES => maps:get(?COMPLIANCE_TYPES, Opts, []),
+        ?CONFIG_RULE_NAMES        => maps:get(?CONFIG_RULE_NAMES, Opts, []),
+        ?NEXT_TOKEN_LABEL         => maps:get(?NEXT_TOKEN_LABEL, Opts, null)
```

Comment:
```
1> jsx:encode(#{<<"NextToken">> => null}).
<<"{\"NextToken\":null}">>
```
@ikemsley have you checked that AWS API handles `null`s? Same question about other optional parameters below.

Reply:
Si, senor.  It handles nulls.